config/routes.rb

```diff
@@ -1,4 +1,13 @@
 Rails.application.routes.draw do
+  root 'welcome#index'
+
+  patch '/movies/:id/upvote' => 'movies#upvote', as: :movie_upvote
+  patch '/books/:id/upvote' => 'books#upvote', as: :book_upvote
+  patch '/albums/:id/upvote' => 'albums#upvote', as: :album_upvote
```

Comment:
Since you have the `resources :whatever` lines below, we can actually add these upvote routes in a slightly simpler way:
```ruby
resources :movies do
  patch 'upvote'
end
```
This will add a route just like the one you've defined above (I think the only different is that `:id` in the path will be `:movie_id` instead).